pkg/console/subresource/route/route.go

```diff
@@ -24,6 +30,49 @@ const (
 	defaultIngressController = "default"
 )
 
+// holds information about custom TLS certificate and its key
+type CustomTLSCert struct {
+	Certificate string
+	Key         string
+}
+
+func ApplyRoute(client routeclient.RoutesGetter, recorder events.Recorder, required *routev1.Route) (*routev1.Route, bool, error) {
+	existing, err := client.Routes(required.Namespace).Get(context.TODO(), required.Name, metav1.GetOptions{})
```

Comment:
This is interesting:

```golang
client.Routes(required.Namespace).Get(context.TODO()
```
The `context.TODO()` throws me.

Reply:
Must be the new norm?

Reply:
It came with kube 1.18 which adopted this from Go:

https://golang.org/pkg/context:
```
Incoming requests to a server should create a Context, and outgoing calls
to servers should accept a Context. The chain of function calls between
them must propagate the Context, optionally replacing it with a derived
Context created using WithCancel, WithDeadline, WithTimeout, or WithValue.
When a Context is canceled, all Contexts derived from it are also canceled. 
```

Reply:
yeah this came from k8s. we can either solve it like this or pass the context as argument to the  `ApplyRoute(ctx context.Context, ...)`